### DISCUSSION TITLE
ASRock B450M HDV R4.0

### DISCUSSION THREAD

**Discussion by Raqnick** [terminatorul/NvStrapsReBar#37] [Q&A] [answered]
Hello , I have an ASRock B450M HDV R4.0 motherboard. There is information in the guide that i need to Enable ResizableBAR ; or Enable Above 4G decoding and disable CSM, but on this motherboard i can't enable BAR without disabling CMS. So i need to disable CSM and Enable ResizableBAR and/or Above 4G Decoding ? Do i need to enable both ?

**Comment by UnidentifiedTag** [comment]
CSM need to be disabled 
If you have secure boot enable that
Above 4g needs to be enabled 

Fast forward to 56:37 in the video : 
[https://youtu.be/ldnC18u8FI0?si=Wj7Oqv2IkUZ_5Mvu](https://youtu.be/ldnC18u8FI0?si=Wj7Oqv2IkUZ_5Mvu)

**Reply by Raqnick** [reply]
Do I need to enable resizeableBAR also ? Or only Above 4g decoding.
Now I have another problem:when I'm trying to flash modded bios it prompts me with "Secure flash check fail" and restarts

**Reply by UnidentifiedTag** [reply, marked as answer]
If you’re motherboard has a rebar option I believe you leave that off, you use the executable you made to turn rebar on and off, so only 4g decoding you should enable 

You’ll have to look up online how to get around asrock’s secure flash check feature, most people use the offline flash, where you format a usb in fat32 and name the file something the manufacturer has told the board to look for when pressing the offline flash button, which you do when the computer is off

it’s documented [here](https://www.asrock.com/support/QA/FlashbackSOP.pdf)

**Reply by Raqnick** [reply]
Hey, thanks for your reply! I've found some guides on this topic on Reddit, but I've decided to not elaborate further because of how complicated this become. Anyway, maybe, I will return to it
